src/parse.c

```diff
@@ -2313,6 +2313,10 @@ handle_bond_primary_member(NetplanParser* npp, yaml_node_t* node, const void* da
                               npp->current.netdef->id, npp->current.netdef->bond_params.primary_member);
 
         ref_ptr = ((char**) ((void*) component + GPOINTER_TO_UINT(data)));
+        if (*ref_ptr) {
+            return yaml_error(npp, node, error, "%s: interface '%s' is already a primary of another bond",
```

Comment:
nitpick (non-blocking): Can we easily give the name of the "other bond" in this log message? That would be nice, but not necessary!

Reply:
Yes, I considered doing that initially but changed my mind to avoid a linear search through all the parsed netdefs. Although I think it's worth it and even if you have lots of netdefs it should be fast...

I added that in a new commit:

```yaml
network:
  version: 2
  ethernets:
    eno1: {}
  bonds:
    bond0:
      parameters:
        primary: eno1
    bond1:
      parameters:
        primary: eno1
```

```
/tmp/fakeroot/etc/netplan/asd.yaml:11:18: Error in network definition: bond1: interface 'eno1' is already a primary of bond0
        primary: eno1
                 ^
```